src/components/booking/partial/steps-actions.tsx

```diff
@@ -0,0 +1,227 @@
+import moment from "moment";
+import { createTimeSlot, updateBookingInfo } from "@/services/notion.service";
+
+type StepAction = {
+  execute:
+  (formData: FormData, boat: Boat) => void
+}
+
+export const steps = [
+  "fuel",
+  "sign",
+  // "validateFront",
+  "uploadFrontIdImage",
+  // "validateBack",
+  "uploadBackIdImage",
+  "pay",
+  "saveData"
+];
+
+const storeIdImage = async (
+  id: string,
+  boatInfo: Boat,
+  file: File,
+  slag: string
+) => {
+  const response = await uploadFile(file, boatInfo.Nombre, id, slag);
+  if (!response.id) {
+    return "";
+  }
+  const url = `https://drive.google.com/file/d/${response.id}/view`;
+  return url;
+};
+
+export const stepsActions = ({
+  setModalInfo,
+  nextStep,
+  booking,
+  bookingId
+}: {
+  setModalInfo: React.Dispatch<React.SetStateAction<{
+    modal: string;
+    message: string;
+    error: string;
+  }>>,
+  nextStep: () => void;
+  booking: Booking,
+  bookingId: string,
+}): Record<string, StepAction> => {
+  var imageFrontLink = ""
+  var imageBackLink = ""
+
+  const fuel = {
+    execute: (formData: FormData, boat: Boat) => {
+      setModalInfo({
+        modal: "loading",
+        message: "Loading fuel modal",
+        error: ""
+      });
+      if (+formData["Fuel Payment"] === 0) {
+        setModalInfo({ modal: "fuel", message: "", error: "" });
+        return;
+      }
+      nextStep();
+    },
+  };
+
+  const sign = {
+    execute: (formData: FormData, boat: Boat) => {
+      setModalInfo({
+        modal: "loading",
+        message: "Loading contract modal",
+        error: ""
+      });
+      if (!formData["signedContract"]) {
+        setModalInfo({ modal: "sign", message: "", error: "" });
+        return
+      }
+      nextStep();
+    },
+  };
+
+  const uploadFrontIdImage = {
+    execute: async (formData: FormData, boat: Boat) => {
+      setModalInfo({
+        modal: "loading",
+        message: "Uploading front image of your identity",
+        error: ""
+      });
+      if (imageFrontLink !== "") {
+        return nextStep();
+      }
+      const uploadIdFrontResponse =
+        await Promise.all([
+          storeIdImage(
+            formData["ID Number"],
+            boat,
+            formData["ID_Front_Picture"] as File,
+            "front"
+          ),
+        ]);
+
+      if (!uploadIdFrontResponse) {
+        toast.error(t("error.upload_image"));
+        setModalInfo({
+          modal: "loading",
+          message: "Uploading front image of your identity",
+          error: "Error uploading front picture",
+        });
+        return;
+      }
+      imageFrontLink = uploadIdFrontResponse[0]
+      nextStep();
+    },
+  }
+
+  const uploadBackIdImage = {
+    execute: async (formData: FormData, boat: Boat) => {
+      setModalInfo({
+        modal: "loading",
+        message: "Uploading back image of your identity",
+        error: ""
+      });
+      if (imageBackLink !== "") {
+        return nextStep();
+      }
```

Comment:
What happen if I uploaded a new image? Shall check the name of the image  more than the URL

But this is the idea yes